packages/carousel/src/Carousel/Carousel.tsx

```diff
@@ -1,9 +1,35 @@
-import React from 'react';
+import React, { Children, isValidElement } from 'react';
 import { IconChevronLeft, IconChevronRight, Flex, IconButton } from '@cmpsr/components';
 import { CarouselProvider, Slider, Slide, ButtonBack, ButtonNext, DotGroup, Dot } from 'pure-react-carousel';
 import { CarouselProps, CarouselButtonProps, DotProps, DotGroupProps, NavigationContainerProps } from './types';
 
-export const Carousel: CarouselProps = (props) => <CarouselProvider {...props} />;
+export const Carousel: CarouselProps = ({ children, showDots = true, showArrows = false, ...props }) => {
+  const [totalSlides] = Children.map(children, (child) => {
+    if (isValidElement(child) && child.type === Carousel.Slider) {
+      return Children.count(child.props.children) as number;
+    }
+  });
+  if (!totalSlides) return;
+
+  return (
+    <CarouselProvider totalSlides={totalSlides} {...props}>
+      <>
+        {children}
+        {(showDots || showArrows) && (
+          <Carousel.NavigationContainer>
+            {showArrows && <Carousel.ButtonBack />}
+            {Array(totalSlides)
+              .fill(0)
+              .map((_, index) => (
+                <Carousel.Dot key={index} slide={index} />
+              ))}
```

Comment:
this is looping twice all the items of the array, can we make it in a single loop instead?